### PR TITLE
fix(manager): use typeof function instead of directly comparison

### PR DESCRIPTION
**Pull request by murka**
Fix [CWE-570](https://cwe.mitre.org/data/definitions/570.html) and [CWE-571](https://cwe.mitre.org/data/definitions/571.html)

More info: https://github.com/murka/outline-server/security/code-scanning/3

### CODE REVIEW

**Review by fortuna — APPROVED** [review]

**Review by fortuna — COMMENTED** [review]
I don't understand, how is that an issue? It's working. The links you sent are for Java?

**Comment by murka** [comment]
It is just example on Java.

This is recommendation: https://codeql.github.com/codeql-query-help/javascript/js-comparison-between-incompatible-types/

File: src/server_manager/web_app/ui_components/outline-server-settings.ts [hunk 1]
```diff
@@ -441,7 +441,7 @@ Polymer({
   _defaultDataLimitEnabledChanged(e: CustomEvent) {
     const wasDataLimitEnabled = this.isDefaultDataLimitEnabled;
     const isDataLimitEnabled = e.detail.value === 'enabled';
-    if (isDataLimitEnabled === undefined || wasDataLimitEnabled === undefined) {
+    if (typeof isDataLimitEnabled === undefined || typeof wasDataLimitEnabled === undefined) {
```
Comment (fortuna) [inline src/server_manager/web_app/ui_components/outline-server-settings.ts:444]: First, there's nothing wrong with testing against undefined with `===` for types that are optional. That is clearer than typeof. The problem is the typing of the variables.

Another issue here is that `isDataLimitEnabled` is guaranteed to be boolean, so this test is bad.

This code would make the types clearer and the code easier to understand:
```ts
if (e.detail?.value === undefined) {
  return;
}
const isDataLimitEnabled = e.detail.value === 'enabled';  // guaranteed to be real Boolean
// This is actually odd, I don't think we need this test. Should probably remove.
// if (this.isDefaultDataLimitEnabled === undefined) {
//  return;
// }
if (isDataLimitEnabled === this.isDefaultDataLimitEnabled) {
  return;
}
this.isDefaultDataLimitEnabled = isDataLimitEnabled;
if (isDataLimitEnabled) {
  this._requestSetDefaultDataLimit();
} else {
  this.fire('RemoveDefaultDataLimitRequested');
}
```

**Review by fortuna — CHANGES_REQUESTED** [review]

File: src/server_manager/web_app/ui_components/outline-server-settings.ts (hunk 1 above)
Comment (murka) [reply src/server_manager/web_app/ui_components/outline-server-settings.ts:444]: LGTM!

**Review by murka — COMMENTED** [review]
All the workflows are passed! The PR is ready for merge, imho :)

https://github.com/murka/outline-server/actions/runs/7652358841

**Review by fortuna — APPROVED** [review]

**Comment by fortuna** [comment]
@daniellacosse another look?

File: src/server_manager/web_app/ui_components/outline-server-settings.ts [hunk 2]
```diff
@@ -439,11 +439,11 @@ Polymer({
...
   _defaultDataLimitEnabledChanged(e: CustomEvent) {
-    const wasDataLimitEnabled = this.isDefaultDataLimitEnabled;
-    const isDataLimitEnabled = e.detail.value === 'enabled';
-    if (isDataLimitEnabled === undefined || wasDataLimitEnabled === undefined) {
+    if (e.detail?.value === undefined) {
       return;
-    } else if (isDataLimitEnabled === wasDataLimitEnabled) {
+    }
+    const isDataLimitEnabled = e.detail.value === 'enabled';
+    if (isDataLimitEnabled === this.isDefaultDataLimitEnabled) {
```
[the remark this answers is not in the document]
Comment (fortuna) [reply src/server_manager/web_app/ui_components/outline-server-settings.ts:446]: Perhaps name it `enableDataLimit`?

```ts
if (this.isDefaultDataLimitEnabled === enableDataLimit) {
  return;
}
```

File: src/server_manager/web_app/ui_components/outline-server-settings.ts [hunk 3]
```diff
@@ -439,11 +439,11 @@ Polymer({
...
       return;
     }
     this.isDefaultDataLimitEnabled = isDataLimitEnabled;
```
Comment (fortuna) [inline src/server_manager/web_app/ui_components/outline-server-settings.ts:449]:
```suggestion
    const enableDataLimit = e.detail.value === 'enabled';
    if (this.isDefaultDataLimitEnabled === enableDataLimit) {
      return;
    }
    this.isDefaultDataLimitEnabled = enableDataLimit;
```

Comment (murka) [reply src/server_manager/web_app/ui_components/outline-server-settings.ts]: LGTM!

**Comment by fortuna** [comment]
@murka it seems like the test failed. Can you take a look?
npm run action server_manager/test

**Pull request merged**